cryptarchia/cryptarchia.py

```diff
@@ -27,24 +28,39 @@ class TimeConfig:
 class Config:
     k: int
     active_slot_coeff: float  # 'f', the rate of occupied slots
+    # The numerator multiplier in the formula used for calculating the number of slots in an epoch.
+    # An epoch is exactly int(floor(epoch_length_multiplier * k / f slots)) long.
+    epoch_length_multiplier: int
```

Comment:
epoch length is derived from the stabilization periods of the different snapshots taken within an epoch.

Thus, the target is  (4 + 3 + 3) * k / f
where 4 is for stabilizing the stake freezing snapshot, 3 is the buffer between frozen stake and nonce snapshot, and the last 3 is for stabilizing the nonce snapshot.

how about we decompose this multiple into those three periods?

```python
    epoch_period_stake_distribution: int
    epoch_period_nonce_buffer: int
    epoch_period_nonce: int

   @property
    def base_period_length(self) -> int:
        return int(floor(self.k / self.active_slot_coeff))

   @property
    def epoch_length(self) -> int:
        return (self.epoch_period_stake_distribution_stabilization
            + self.epoch_period_nonce_buffer
            + self.epoch_period_nonce_stabilization) * self.epoch_period
```

Then we can use these period configs when creating snapshots

Reply:
Makes sense, and makes it impossible to configure values that do not make sense (e.g. seting `epoch_length_multiplier` to say 2 when the nonce snapshot is taken after 7k/f)